docs/content/NavList.mdx

```diff
@@ -0,0 +1,523 @@
+    <NavList.LeadingVisual>#️⃣</NavList.LeadingVisual>
+    General
+  </NavList.Item>
+  <NavList.Item href="/q-a">
+    <NavList.LeadingVisual>🙏</NavList.LeadingVisual>
+    Q&A
+  </NavList.Item>
+  <NavList.Item href="/show-and-tell">
+    <NavList.LeadingVisual>🙌</NavList.LeadingVisual>
+    Show and tell
+  </NavList.Item>
+</NavList>
+```
+
+<details>
+  <summary>Rendered HTML</summary>
+
+```html
+<nav>
+  <ul role="list">
+    <li>
+      <a href="/" aria-current="page">
+        <span>#️⃣</span>
+        <span>General</span>
+      </a>
+    </li>
+    <li>
+      <a href="/pulls">
+        <span>🙏</span>
+        <span>Q&A</span>
+      </a>
+    </li>
+    <li>
+      <a href="/issues">
+        <span>🙌</span>
+        <span>Show and tell</span>
+      </a>
+    </li>
+  </ul>
+</nav>
+```
+
+</details>
+
+### With trailing visuals
+
+```jsx
+<NavList>
+  <NavList.Item href="/inbox" aria-current="page">
+    Inbox
+    <NavList.TrailingVisual>1,234</NavList.TrailingVisual>
+  </NavList.Item>
+  <NavList.Item href="/saved">Saved</NavList.Item>
+  <NavList.Item href="/done">Done</NavList.Item>
+</NavList>
+```
+
+<details>
+  <summary>Rendered HTML</summary>
+
+```html
+<nav>
+  <ul role="list">
+    <li>
+      <a href="/inbox" aria-current="page">
+        <span>Inbox</span>
+        <span>1,234</span>
+      </a>
+    </li>
+    <li><a href="/saved">Saved</a></li>
+    <li><a href="/done">Done</a></li>
+  </ul>
+</nav>
+```
+
+</details>
+
+### With a heading
+
+```jsx
+<>
+  <h3 id="workflows-heading">Workflows</h3>
+  <NavList aria-labelledby="workflows-heading">
+    <NavList.Item href="/" aria-current="page">
+      All workflows
+    </NavList.Item>
+    <NavList.Item href="/ci">CI</NavList.Item>
+    <NavList.Item href="/deploy">Deploy</NavList.Item>
+    <NavList.Item href="/release">Release</NavList.Item>
+  </NavList>
+</>
+```
+
+<details>
+  <summary>Rendered HTML</summary>
+
+```html
+<h3 id="workflows-heading">Workflows</h3>
+<nav aria-labelledby="workflows-heading">
+  <ul role="list">
+    <li><a href="/" aria-current="page">All workflows</a></li>
+    <li><a href="/ci">CI</a></li>
+    <li><a href="/deploy">Deploy</a></li>
+    <li><a href="/release">Release</a></li>
+  </ul>
+</nav>
+```
+
+</details>
+
+### With aria-label
+
+```jsx
+<NavList aria-label="Security">
+  <NavList.Item href="/" aria-current="page">
+    Overview
+  </NavList.Item>
+  <NavList.Item href="/policy">Security policy</NavList.Item>
+  <NavList.Item href="/advisories">Security advisories</NavList.Item>
+</NavList>
+```
+
+<details>
+  <summary>Rendered HTML</summary>
+
+```html
+<nav aria-label="Security">
+  <ul role="list">
+    <li><a href="/" aria-current="page">Overview</a></li>
+    <li><a href="/policy">Security policy</a></li>
+    <li><a href="/advisories">Security advisories</a></li>
+  </ul>
+</nav>
+```
+
+</details>
+
+### With groups
+
+```jsx
+<NavList>
+  <NavList.Group title="Overview">
```

Comment:
I assume groups in `NavList` are analogous to the concept of sections in `NavigationList`? If so, why not call them sections in `NavList` as well?

Reply:
I called it `Group` because that what it's called in`ActionList`: https://primer.style/react/ActionList#with-groups

It looks like "section" and "group" are used interchangeably in [some of our docs](https://primer.style/css/components/action-list#divider-with-label-text). I propose that we standardize on "group" in Primer React and ViewComponents because it aligns with the ARIA role ([group](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/group_role)). Here's some [more context](https://github.com/github/primer/issues/650#issuecomment-1029340402)